padertorch/contrib/examples/tasnet/tas_coders.py

```diff
@@ -127,3 +129,90 @@ def forward(self, w: torch.Tensor) -> torch.Tensor:
         """
         # Convolution from (B, N, T_enc) to (B, 1, T) and remove channel
         return self.decoder_1d(w)[:, 0, :]
+
+
+class StftEncoder(torch.nn.Module):
+    """
+    >>> mixture = torch.rand((2, 6, 203))
+    >>> stft_encoder = StftEncoder(N=258)
+    >>> encoded, num_frames = stft_encoder(mixture, [203, 150])
+    >>> encoded.shape
+    torch.Size([2, 6, 258, 20])
+    >>> num_frames
+    [20, 14]
+    >>> from paderbox.transform import stft
+    >>> import numpy as np
+    >>> stft_out = stft(\
+            mixture.numpy(), 256, 10, window_length=20, \
+            fading=None)
+    >>> stft_encoded = np.concatenate(\
+            [np.real(stft_out), np.imag(stft_out)], axis=-1\
+        ).transpose(0,1,3,2)
+    >>> np.testing.assert_allclose(stft_encoded, encoded, atol=1e-5)
+    """
+    def __init__(self, L: int = 20, N: int = 256, stride: int = None):
```

Comment:
I don't like `L` and `N`. Do you have a better name for these variables? I think they will now also appear in the config, so a longer name would be better.

Reply:
I can change them to window_size and num_feats but in that case they should also be changed for the encoder and decoder

Reply:
Since encoder and decoder is hidden for the user at the moment, we can change them too (and should do, to keep them consistent).

But as always: No shortcuts in the signature: `num_feats` -> `num_features`

Reply:
I used num_feats since it is used in tasnet and dprnn but I can change that as well

Reply:
I didn't check the others. Use the name that you think is better.